src/client.js

```diff
@@ -57,7 +59,29 @@ class SkynetClient {
   }
 }
 
-module.exports = { SkynetClient };
+/**
+ * Helper function that builds the request headers.
+ *
+ * @param [headers] - Any base headers.
+ * @param [customUserAgent] - A custom user agent to set.
+ * @param [customCookie] - A custom cookie.
+ * @returns - The built headers.
+ */
+function buildRequestHeaders(headers, customUserAgent, customCookie) {
+  if (!headers) {
+    headers = {};
+  }
+  // Set some headers from common options.
+  if (customUserAgent) {
+    headers["User-Agent"] = customUserAgent;
+  }
+  if (customCookie) {
+    headers["Cookie"] = customCookie;
+  }
+  return headers;
+}
```

Comment:
you're mutating `headers` if they are passed as an argument - you should assign to new object